build.gradle

```diff
@@ -28,20 +28,20 @@ task sourcesJar(type: Jar, dependsOn: classes) {
 }
 
 dependencies {
-    compile(
+    implementation(
             'junit:junit:4.12',
             'org.elasticsearch.client:transport:6.2.2',
             'org.apache.logging.log4j:log4j-core:2.8.2'
     )
 
-    testCompile(
+    testImplementation(
             'junit:junit:4.12',
             'org.spockframework:spock-core:1.1-groovy-2.4',
             'org.mockito:mockito-core:2.8.47')
 }
 
 configurations.all {
     resolutionStrategy {
-        force 'org.codehaus.groovy:groovy-all:2.4.12'
+        force 'org.codehaus.groovy:groovy-all:2.5.12'
```

Comment:
I have to double check it but I think groovy 3 is required for gradle 7 compatibility

Reply:
Done.